src/libraries/JANA/Services/JParameterManager.h

```diff
@@ -369,12 +373,39 @@ template<typename T, size_t N>
 inline void JParameterManager::Parse(const std::string& value,std::array<T,N> &val) {
     std::string s;
     std::stringstream ss(value);
+    std::string temp = "";   // creating a temp var allows us to store the string s where the escape character was used
     int indx = 0;
     while (getline(ss, s, ',')) {
         T t;
         Parse(s, t);
         val[indx++]= t;
-    }    
+    }
+}
+
+// @brief Template to parse a string and return in an array of strings
+template<size_t N>
+inline void JParameterManager::Parse(const std::string& value,std::array<std::string,N> &val) {
+    std::string s;
+    std::stringstream ss(value);
+    std::string temp = "";   // creating a temp var allows us to store the string s where the escape character was used
+    int indx = 0;
+    while (getline(ss, s, ',')) {
+        std::string t;
+        if (s.find('\\') != std::string::npos) {
+            s.erase(s.find('\\'));
+            temp += s + ',';
```

Comment:
When find yourself concatenating a string inside a loop, you should probably be using a string builder instead, because it will have much better time complexity. In C++ you would use `std::ostringstream`

Reply:
Never mind, this might not be a problem in std::string. It certainly was a problem in other contexts such as java's String class